src/main/capi/duckdb_value-c.cpp

```diff
@@ -34,6 +34,137 @@ duckdb_value duckdb_create_int64(int64_t input) {
+}
+duckdb_value duckdb_create_smallint(int16_t input) {
+	return WrapValue(new duckdb::Value(duckdb::Value::SMALLINT(input)));
+}
+int16_t duckdb_get_smallint(duckdb_value val) {
+	return UnwrapValue(val).GetValue<int16_t>();
+}
+duckdb_value duckdb_create_usmallint(uint16_t input) {
+	return WrapValue(new duckdb::Value(duckdb::Value::USMALLINT(input)));
+}
+uint16_t duckdb_get_usmallint(duckdb_value val) {
+	return UnwrapValue(val).GetValue<uint16_t>();
+}
+duckdb_value duckdb_create_integer(int32_t input) {
+	return WrapValue(new duckdb::Value(duckdb::Value::INTEGER(input)));
+}
+int32_t duckdb_get_integer(duckdb_value val) {
+	return UnwrapValue(val).GetValue<int32_t>();
+}
+duckdb_value duckdb_create_uinteger(uint32_t input) {
+	return WrapValue(new duckdb::Value(duckdb::Value::UINTEGER(input)));
+}
+uint32_t duckdb_get_uinteger(duckdb_value val) {
+	return UnwrapValue(val).GetValue<uint32_t>();
+}
+duckdb_value duckdb_create_ubigint(uint64_t input) {
+	return WrapValue(new duckdb::Value(duckdb::Value::UBIGINT(input)));
+}
+uint64_t duckdb_get_ubigint(duckdb_value val) {
+	return UnwrapValue(val).GetValue<uint64_t>();
+}
+duckdb_value duckdb_create_bigint(int64_t input) {
+	return WrapValue(new duckdb::Value(duckdb::Value::BIGINT(input)));
+}
+int64_t duckdb_get_bigint(duckdb_value val) {
+	return UnwrapValue(val).GetValue<int64_t>();
+}
+duckdb_value duckdb_create_float(float input) {
+	return WrapValue(new duckdb::Value(duckdb::Value::FLOAT(input)));
+}
+float duckdb_get_float(duckdb_value val) {
+	return UnwrapValue(val).GetValue<float>();
+}
+duckdb_value duckdb_create_double(double input) {
+	return WrapValue(new duckdb::Value(duckdb::Value::DOUBLE(input)));
+}
+double duckdb_get_double(duckdb_value val) {
+	return UnwrapValue(val).GetValue<double>();
+}
+duckdb_value duckdb_create_date(duckdb_date input) {
+	return WrapValue(new duckdb::Value(duckdb::Value::DATE(duckdb::date_t(input.days))));
+}
+duckdb_date duckdb_get_date(duckdb_value val) {
+	auto res = UnwrapValue(val).GetValue<duckdb::date_t>();
+	return {res.days};
+}
+duckdb_value duckdb_create_time(duckdb_time input) {
+	return WrapValue(new duckdb::Value(duckdb::Value::TIME(duckdb::dtime_t(input.micros))));
+}
+duckdb_time duckdb_get_time(duckdb_value val) {
+	auto dtime = UnwrapValue(val).GetValue<duckdb::dtime_t>();
+	return {dtime.micros};
+}
+duckdb_value duckdb_create_time_tz_value(duckdb_time_tz value) {
+	return WrapValue(new duckdb::Value(duckdb::Value::TIMETZ(duckdb::dtime_tz_t(value.bits))));
+}
+duckdb_time_tz duckdb_get_time_tz(duckdb_value val) {
+	auto time_tz = UnwrapValue(val).GetValue<duckdb::dtime_tz_t>();
+	return {time_tz.bits};
+}
+duckdb_value duckdb_create_timestamp(duckdb_timestamp input) {
+	return WrapValue(new duckdb::Value(duckdb::Value::TIMESTAMP(duckdb::timestamp_t(input.micros))));
+}
+duckdb_timestamp duckdb_get_timestamp(duckdb_value val) {
+	auto timestamp = UnwrapValue(val).GetValue<duckdb::timestamp_t>();
```

Comment:
`GetValue` can throw an exception in case the conversion cannot be performed - so we need a `try/catch` in these functions. Could we add tests with these functions used on "wrong" types (e.g. `duckdb_get_timestamp(duckdb_create_smallint(10))`)?